activerecord/test/cases/associations_test.rb

```diff
@@ -461,6 +461,29 @@ def test_nullify_composite_has_many_through_association
     assert_empty(blog_post.reload.tags)
     assert_not_predicate Sharded::BlogPostTag.where(blog_post_id: blog_post.id, blog_id: blog_post.blog_id), :exists?
   end
+
+  def test_using_query_constraints_warns_about_changing_behavior
+    has_many_expected_message = <<~MSG.squish
+      Setting `query_constraints:` option on `Sharded::BlogPost.has_many :qc_deprecated_comments` is deprecated
+      and its behavior may be modifed in Rails 8.
+      To preserve the current behavior please use the `foreign_key` option instead.
+    MSG
```

Comment:
We don't enforce this style but generally prefer a new line between MSG and the assertion. It makes the test easier to read.

Reply:
100% agree. That was an overlook on my side, will change